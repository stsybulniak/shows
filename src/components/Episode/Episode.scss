@import "../../assets/scss/screenVariables.scss";

.Episode {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &__backButton {
    background-color: var(--color-primary);
    padding: 1rem;
    cursor: pointer;
    margin-right: auto;
    color: white;
    border-radius: .5rem;
    margin-bottom: 2rem;
  }

  img {
    margin-top: 3rem;
    max-width: 120rem;
  }

  &__description {
    margin-top: 3rem;
    font-size: 1.6rem;
    max-width: 120rem;
  }

  @media only screen and (max-width: $bp-large) {
    img {
      width: 90%;
    }
    &__description {
      width: 90%;
    }
  }
}