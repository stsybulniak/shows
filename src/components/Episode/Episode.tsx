import { FC } from 'react';
import { IEpisode } from '../../store/episode';
import { useHistory } from 'react-router-dom';
import './Episode.scss';
import placeholderImage from '../../assets/img/test.png';
import { BREAKPOINT_TO_USE_MEDIUM_IMG } from '../../utils/constants';

interface IEpisodeProps {
  episode: IEpisode;
  viewportWidth: number;
}

const Episode: FC<IEpisodeProps> = ({ episode, viewportWidth }) => {
  const history = useHistory();

  return (
    <div className='Episode'>
      <button className='Episode__backButton' onClick={() => history.push('/')}>
        Back to list
      </button>
      <h1>{episode.name}</h1>
      <img
        src={
          (viewportWidth > BREAKPOINT_TO_USE_MEDIUM_IMG ? episode.image?.original : episode.image?.medium) ||
          placeholderImage
        }
        alt={episode.name}
      />
      <div className='Episode__description' dangerouslySetInnerHTML={{ __html: episode.summary }} />
    </div>
  );
};

export default Episode;
