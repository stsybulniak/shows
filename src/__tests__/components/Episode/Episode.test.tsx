import { fireEvent, render, screen } from '@testing-library/react';
import Episode from '../../../components/Episode/Episode';

const episodeMock = {
  id: 1156862,
  name: 'Mini Golf Madness',
  season: 2,
  number: 8,
  image: {
    medium: 'https://static.tvmaze.com/uploads/images/medium_landscape/115/288994.jpg',
    original: 'https://static.tvmaze.com/uploads/images/original_untouched/115/288994.jpg',
  },
  summary: '<p>Buttercup takes her game.</p>',
};

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<Episode />', () => {
  it('Episode matches snapshot with origin image', () => {
    const { baseElement } = render(<Episode episode={episodeMock} viewportWidth={1200} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Episode matches snapshot with medium image', () => {
    const { baseElement } = render(<Episode episode={episodeMock} viewportWidth={640} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Episode test click with Router trigger', async () => {
    render(<Episode episode={episodeMock} viewportWidth={640} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
