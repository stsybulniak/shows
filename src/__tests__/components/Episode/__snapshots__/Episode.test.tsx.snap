// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episode /> Episode matches snapshot with medium image 1`] = `
<body>
  <div>
    <div
      class="Episode"
    >
      <button
        class="Episode__backButton"
      >
        Back to list
      </button>
      <h1>
        Mini Golf Madness
      </h1>
      <img
        alt="Mini Golf Madness"
        src="https://static.tvmaze.com/uploads/images/medium_landscape/115/288994.jpg"
      />
      <div
        class="Episode__description"
      >
        <p>
          Buttercup takes her game.
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Episode /> Episode matches snapshot with origin image 1`] = `
<body>
  <div>
    <div
      class="Episode"
    >
      <button
        class="Episode__backButton"
      >
        Back to list
      </button>
      <h1>
        Mini Golf Madness
      </h1>
      <img
        alt="Mini Golf Madness"
        src="https://static.tvmaze.com/uploads/images/original_untouched/115/288994.jpg"
      />
      <div
        class="Episode__description"
      >
        <p>
          Buttercup takes her game.
        </p>
      </div>
    </div>
  </div>
</body>
`;
